File: app/src/main/java/com/amaze/filemanager/database/daos/SftpEntryDao.java
package com.amaze.filemanager.database.daos;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.amaze.filemanager.database.models.utilities.SftpEntry;

import static com.amaze.filemanager.database.UtilitiesDatabase.COLUMN_HOST_PUBKEY;
import static com.amaze.filemanager.database.UtilitiesDatabase.COLUMN_NAME;
import static com.amaze.filemanager.database.UtilitiesDatabase.COLUMN_PATH;
import static com.amaze.filemanager.database.UtilitiesDatabase.COLUMN_PRIVATE_KEY;
import static com.amaze.filemanager.database.UtilitiesDatabase.COLUMN_PRIVATE_KEY_NAME;
import static com.amaze.filemanager.database.UtilitiesDatabase.TABLE_SFTP;

@Dao
public interface SftpEntryDao {

    @Insert
    public void insert(SftpEntry instance);

    @Update
    public void update(SftpEntry instance);

    @Query("SELECT * FROM " + TABLE_SFTP)
    public SftpEntry[] list();

    @Query("SELECT * FROM " + TABLE_SFTP + " WHERE " + COLUMN_NAME + " = :name AND " + COLUMN_PATH + " = :path")
    public SftpEntry findByNameAndPath(String name, String path);

    @Query("SELECT * FROM " + TABLE_SFTP + " WHERE " + COLUMN_NAME + " = :name")
    public SftpEntry findByName(String name);

    @Query("SELECT " + COLUMN_HOST_PUBKEY + " FROM " + TABLE_SFTP + " WHERE PATH = :uri")
    public String getSshHostKey(String uri);

    @Query("SELECT " + COLUMN_PRIVATE_KEY_NAME + " FROM " + TABLE_SFTP + " WHERE PATH = :uri")
    public String getSshAuthPrivateKeyName(String uri);

    @Query("SELECT " + COLUMN_PRIVATE_KEY + " FROM " + TABLE_SFTP + " WHERE PATH = :uri")
    public String getSshAuthPrivateKey(String uri);

    @Query("DELETE FROM " + TABLE_SFTP + " WHERE " + COLUMN_NAME + " = :name")
    public void deleteByName(String name);

    @Query("DELETE FROM " + TABLE_SFTP + " WHERE " + COLUMN_NAME + " = :name AND " + COLUMN_PATH + " = :path")
    public void deleteByNameAndPath(String name, String path);
}

File: app/src/main/java/com/amaze/filemanager/database/daos/SftpEntryDao.kt
/*
 * Copyright (C) 2014-2023 Arpit Khurana <arpitkh96@gmail.com>, Vishal Nehra <vishalmeham2@gmail.com>,
 * Emmanuel Messulam<emmanuelbendavid@gmail.com>, Raymond Lai <airwave209gt at gmail.com>,
 * Oleksandr Narvatov <hipi96222@gmail.com> and Contributors.
 *
 * This file is part of Amaze File Manager.
 *
 * Amaze File Manager is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.amaze.filemanager.database.daos

import androidx.room.*
import com.amaze.filemanager.database.models.utilities.SftpEntry
import io.reactivex.Completable
import com.amaze.filemanager.database.UtilitiesDatabase
import io.reactivex.Single

/**
 * [Dao] interface definition for [SftpEntry]. Concrete class is generated by Room
 * during build.
 *
 * @see Dao
 *
 * @see SftpEntry
 *
 * @see com.amaze.filemanager.database.UtilitiesDatabase
 */
@Dao
interface SftpEntryDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(instance: SftpEntry): Completable

    @Update
    fun update(instance: SftpEntry): Completable

    @Query("SELECT * FROM " + UtilitiesDatabase.TABLE_SFTP)
    fun list(): Single<List<SftpEntry>>

    @Query(
        "SELECT * FROM "
                + UtilitiesDatabase.TABLE_SFTP
                + " WHERE "
                + UtilitiesDatabase.COLUMN_NAME
                + " = :name AND "
                + UtilitiesDatabase.COLUMN_PATH
                + " = :path"
    )
    fun findByNameAndPath(name: String, path: String): Single<SftpEntry>

    @Query("SELECT * FROM " + UtilitiesDatabase.TABLE_SFTP + " WHERE " + UtilitiesDatabase.COLUMN_NAME + " = :name")
    fun findByName(name: String): Single<SftpEntry>

    @Query("SELECT " + UtilitiesDatabase.COLUMN_HOST_PUBKEY + " FROM " + UtilitiesDatabase.TABLE_SFTP + " WHERE " + UtilitiesDatabase.COLUMN_PATH + " = :uri")
    fun getRemoteHostKey(uri: String): Single<String?>

    @Query(
        "SELECT "
                + UtilitiesDatabase.COLUMN_PRIVATE_KEY_NAME
                + " FROM "
                + UtilitiesDatabase.TABLE_SFTP
                + " WHERE "
                + UtilitiesDatabase.COLUMN_PATH
                + " = :uri"
    )
    fun getSshAuthPrivateKeyName(uri: String): Single<String?>

    @Query("SELECT " + UtilitiesDatabase.COLUMN_PRIVATE_KEY + " FROM " + UtilitiesDatabase.TABLE_SFTP + " WHERE " + UtilitiesDatabase.COLUMN_PATH + " = :uri")
    fun getSshAuthPrivateKey(uri: String): Single<String?>

    @Query("DELETE FROM " + UtilitiesDatabase.TABLE_SFTP + " WHERE " + UtilitiesDatabase.COLUMN_NAME + " = :name")
    fun deleteByName(name: String): Completable

    @Query(
        "DELETE FROM "
                + UtilitiesDatabase.TABLE_SFTP
                + " WHERE "
                + UtilitiesDatabase.COLUMN_NAME
                + " = :name AND "
                + UtilitiesDatabase.COLUMN_PATH
                + " = :path"
    )
    fun deleteByNameAndPath(name: String, path: String): Completable

}